app/models/miq_request_workflow.rb

```diff
@@ -95,9 +96,13 @@ def create_request(values, _requester = nil, auto_approve = false)
 
     request.log_request_success(@requester, :created)
 
-    request.call_automate_event_queue("request_created")
-    request.approve(@requester, "Auto-Approved") if auto_approve == true
-    request.reload if auto_approve
+    if request.process_immediately_on_create?
+      # TODO: address auto-approve potential issue
+      request.call_automate_event_queue("request_created")
+      request.approve(@requester, "Auto-Approved") if auto_approve == true
```

Comment:
@tinaafitz If we are not processing the event at this time we will lose the `auto_approve' flag.  We need to retain the flag so we can apply it when the service_order is processed.